test/unit/snippets.cpp

```diff
@@ -280,6 +280,32 @@ parsing_urls()
         //[snippet_parsing_url_1
         result< url > ru = parse_uri_reference( "https://www.example.com/path/to/file.txt" );
         if ( ru.has_value() )
+        {
+            url u = ru.value();
+            assert(u.encoded_path() == "/path/to/file.txt");
+        }
+        //]
+        boost::ignore_unused(ru);
+    }
+
```

Comment:
We need all of the javadocs in parse.hpp in here, correctly labeled

Reply:
The javadocs should also go in `snippets.cpp`? I thought `snippets.cpp` were just for the exposition and javadocs should go in `parse.cpp`.

Reply:
Oh... I thought this was a javadoc.